### PR TITLE
Update CMakeLists to reflect gcp upgrade

### PR DESCRIPTION
**Pull request by GuanLuo**

### CODE REVIEW

**Review by nv-kmcgill53 — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -454,7 +454,7 @@ if(${TRITON_ENABLE_GCS})
   target_link_libraries(
     triton-core
     PRIVATE
-      storage_client
+    google-cloud-cpp::storage
```
Comment (jbkyang-nvi) [inline src/CMakeLists.txt:457]: nit: two spaces. see
```
    PRIVATE
      dl
      numa
```

**Review by jbkyang-nvi — APPROVED** [review]

File: src/test/response_cache_test.cc [hunk 2]
```diff
@@ -182,6 +182,17 @@ InferenceResponse::InferenceResponse(
...
+std::ostream&
+operator<<(std::ostream& out, const InferenceResponse& response)
+{
+  out << "[0x" << std::addressof(response) << "] "
+      << "response id: " << response.Id() << std::endl;
+
+  out << "status:" << response.ResponseStatus().AsString() << std::endl;
+
+  return out;
+}
```
Comment (rmccorm4) [inline src/test/response_cache_test.cc:194]: What was this change needed for?

**Review by rmccorm4 — APPROVED** [review]

File: src/test/response_cache_test.cc (hunk 2 above)
Comment (GuanLuo) [reply src/test/response_cache_test.cc:194]: Seems like newer GTest will try to use this outstream operator, so need to provide implementation.

**Review by nv-kmcgill53 — APPROVED** [review]
